### PR TITLE
TOUR-5 : Add the ability to have a tour that change the current page

### PR DESCRIPTION
**Pull request by acotiuga**
* improved the code of steps' creation

### CODE REVIEW

File: application-tour-ui/src/main/resources/TourCode/TourJS.xml [hunk 1]
```diff
@@ -257,15 +257,12 @@ require(['jquery', 'xwiki-meta'], function ($, xm) {
...
-          title     : jsonData.steps[i].title,
-          content   : jsonData.steps[i].content,
-          placement : jsonData.steps[i].placement,
-          backdrop  : jsonData.steps[i].backdrop
+      for (var i = 0; i &lt; jsonData.steps.length; i++) {
+        var step = {};
+        $.each(jsonData.steps[i], function(key, value) {
+          step[key] = value;
         });
+        tour.addStep(step);
```
Comment (mflorea) [inline application-tour-ui/src/main/resources/TourCode/TourJS.xml]: Or even shorter:

    tour.addStep($.extend({}, jsonData.steps[i]))

Comment (acotiuga) [inline application-tour-ui/src/main/resources/TourCode/TourJS.xml]: Done in 043c4f9bcf36cfa818880c6736dbb453d189a44d. Thanks!

Comment (gdelhumeau) [inline application-tour-ui/src/main/resources/TourCode/TourJS.xml]: I have modified it to `tour.addStep(jsonData.steps[i])`. Any objection Marius?

Comment (mflorea) [inline application-tour-ui/src/main/resources/TourCode/TourJS.xml]: @gdelhumeau no objection, as long as you know that ``jsonData.steps`` is not being modified in some other place (that would trigger side effects on the tour).

**Pull request merged**